core/server/data/migrations/versions/3.12/01-remove-legacy-is-paid-flag-from-settings.js

```diff
@@ -0,0 +1,110 @@
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            debug('before cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            const hasIsPaid = Object.prototype.hasOwnProperty.call(subscriptionSettings, 'isPaid');
+
+            if (hasIsPaid) {
+                debug('Removing legacy isPaid flag from members settings');
+                delete subscriptionSettings.isPaid;
+            }
+
+            debug('after cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            return localOptions
+                .transacting('settings')
+                .where('key', settingsKey)
+                .update({
+                    value: JSON.stringify(subscriptionSettings)
+                });
+        });
+};
+
+module.exports.down = (options) => {
+    let localOptions = _.merge({
+        context: {internal: true}
+    }, options);
+    const settingsKey = 'members_subscription_settings';
+
+    return localOptions
+        .transacting('settings')
+        .then((response) => {
+            if (!response) {
+                common.logging.warn('Cannot find settings.');
+                return;
+            }
+
+            let subscriptionSettingsEntry = response.find((entry) => {
+                return entry.key === settingsKey;
+            });
+
+            if (!subscriptionSettingsEntry) {
+                common.logging.warn('Cannot find members subscription settings.');
+                return;
+            }
+
+            let subscriptionSettings = JSON.parse(subscriptionSettingsEntry.value);
+
+            debug('before cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            let isPaid = false;
+
+            const stripePaymentProcessor = subscriptionSettings.paymentProcessors.find(
+                paymentProcessor => paymentProcessor.adapter === 'stripe'
+            );
+
+            if (stripePaymentProcessor && stripePaymentProcessor.config.public_token && stripePaymentProcessor.config.secret_token) {
+                isPaid = true;
+            }
+
+            subscriptionSettings.isPaid = isPaid;
+
+            debug('after cleanup');
+            debug(JSON.stringify(subscriptionSettings, null, 2));
+
+            return localOptions
+                .transacting('settings')
+                .where('key', settingsKey)
+                .update({
+                    value: JSON.stringify(subscriptionSettings)
+                });
```

Comment:
@gargol Though not exactly the same usecase, probably a good idea to set the settings cache here as well to tackle the reboot/in-memory settings cache problem :)

Reply:
Nah, in this situation the variable isn't even used. So no need to worry about that ;)